@WilliamTambellini 

There are some errors building examples:
`python3 scripts/build_cpp_examples.py`

```
[2025-07-01T03:21:23.815Z] /usr/bin/c++ -DENABLE_BF16 -DENABLE_FP8 -I/home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/../../../cpp/include -I/opt/hpcx/ompi/include -I/opt/hpcx/ompi/include/openmpi -I/opt/hpcx/ompi/include/openmpi/opal/mca/hwloc/hwloc201/hwloc/include -I/opt/hpcx/ompi/include/openmpi/opal/mca/event/libevent2022/libevent -I/opt/hpcx/ompi/include/openmpi/opal/mca/event/libevent2022/libevent/include -I/home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/3rdparty/cxxopts/include -isystem /usr/local/cuda/targets/x86_64-linux/include -Wall -pthread -lstdc++ -DENABLE_MULTI_DEVICE=CMakeFiles/executorExampleAdvancedMultiInstances.dir/compiler_depend.tsBOOL:ON> -O3 -std=gnu++17 -D_GLIBCXX_USE_CXX11_ABI=1 -MD -MT CMakeFiles/executorExampleAdvancedMultiInstances.dir/executorExampleAdvancedMultiInstances.cpp.o -MF CMakeFiles/executorExampleAdvancedMultiInstances.dir/executorExampleAdvancedMultiInstances.cpp.o.d -o CMakeFiles/executorExampleAdvancedMultiInstances.dir/executorExampleAdvancedMultiInstances.cpp.o -c /home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/executorExampleAdvancedMultiInstances.cpp
[2025-07-01T03:21:23.815Z] <command-line>: error: token "." is not valid in preprocessor expressions
[2025-07-01T03:21:23.815Z] /home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/../../../cpp/include/tensorrt_llm/runtime/utils/multiDeviceUtils.h:22:5: note: in expansion of macro ‘ENABLE_MULTI_DEVICE’
[2025-07-01T03:21:23.815Z]    22 | #if ENABLE_MULTI_DEVICE
[2025-07-01T03:21:23.815Z]       |     ^~~~~~~~~~~~~~~~~~~
[2025-07-01T03:21:23.815Z] <command-line>: error: token "." is not valid in preprocessor expressions
[2025-07-01T03:21:23.815Z] /home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/../../../cpp/include/tensorrt_llm/runtime/utils/mpiUtils.h:40:5: note: in expansion of macro ‘ENABLE_MULTI_DEVICE’
[2025-07-01T03:21:23.815Z]    40 | #if ENABLE_MULTI_DEVICE
[2025-07-01T03:21:23.815Z]       |     ^~~~~~~~~~~~~~~~~~~
[2025-07-01T03:21:23.815Z] <command-line>: error: token "." is not valid in preprocessor expressions
[2025-07-01T03:21:23.815Z] /home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/../../../cpp/include/tensorrt_llm/runtime/utils/mpiUtils.h:229:5: note: in expansion of macro ‘ENABLE_MULTI_DEVICE’
[2025-07-01T03:21:23.815Z]   229 | #if ENABLE_MULTI_DEVICE
[2025-07-01T03:21:23.815Z]       |     ^~~~~~~~~~~~~~~~~~~
[2025-07-01T03:21:23.815Z] <command-line>: error: token "." is not valid in preprocessor expressions
[2025-07-01T03:21:23.815Z] /home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/../../../cpp/include/tensorrt_llm/runtime/utils/mpiUtils.h:239:5: note: in expansion of macro ‘ENABLE_MULTI_DEVICE’
[2025-07-01T03:21:23.815Z]   239 | #if ENABLE_MULTI_DEVICE
[2025-07-01T03:21:23.815Z]       |     ^~~~~~~~~~~~~~~~~~~
[2025-07-01T03:21:23.815Z] <command-line>: error: token "." is not valid in preprocessor expressions
[2025-07-01T03:21:23.815Z] /home/jenkins/agent/workspace/LLM/helpers/Build-x86_64/llm/examples/cpp/executor/../../../cpp/include/tensorrt_llm/runtime/utils/mpiUtils.h:385:5: note: in expansion of macro ‘ENABLE_MULTI_DEVICE’
[2025-07-01T03:21:23.815Z]   385 | #if ENABLE_MULTI_DEVICE
[2025-07-01T03:21:23.815Z]       |     ^~~~~~~~~~~~~~~~~~~
[2025-07-01T03:21:25.688Z] gmake[2]: *** [CMakeFiles/executorExampleAdvancedMultiInstances.dir/build.make:79: CMakeFiles/executorExampleAdvancedMultiInstances.dir/executorExampleAdvancedMultiInstances.cpp.o] Error 1
```

--- a/examples/cpp/executor/CMakeLists.txt
+++ b/examples/cpp/executor/CMakeLists.txt
@@ -22,10 +22,19 @@ if(NOT TRTLLM_BUILD_DIR)
   set(TRTLLM_BUILD_DIR "${TRTLLM_DIR}/cpp/build")
 endif()
 set(TRTLLM_LIB_PATH "${TRTLLM_BUILD_DIR}/tensorrt_llm/libtensorrt_llm.so")
+if(NOT EXISTS ${TRTLLM_LIB_PATH})
+  message(FATAL_ERROR "Cannot find ${TRTLLM_LIB_PATH}")
+endif()
+
 set(TRTLLM_PLUGIN_PATH
     "${TRTLLM_BUILD_DIR}/tensorrt_llm/plugins/libnvinfer_plugin_tensorrt_llm.so"
 )
 set(TRTLLM_INCLUDE_DIR "${TRTLLM_DIR}/cpp/include")
+
+option(
+  ENABLE_MULTI_DEVICE
+  "Enable multi device/instance examples building (requires MPI headers/libs)"
+  ON)
 
 # Determine CXX11 ABI compatibility
 execute_process(
@@ -47,8 +56,13 @@ set(CMAKE_VERBOSE_MAKEFILE 1)
 # Define project name
 project(executorExamples)
 
-# Compile options
-set(CMAKE_CXX_FLAGS "-Wall -pthread -lstdc++ -DENABLE_MULTI_DEVICE=1")
+# Compile options $<BOOL:${ENABLE_MULTI_DEVICE}> ?
+if(ENABLE_MULTI_DEVICE)
+  set(EMD 1)
+else()
+  set(EMD 0)
+endif()
+set(CMAKE_CXX_FLAGS "-Wall -pthread -lstdc++ -DENABLE_MULTI_DEVICE=${EMD} ")
 set(CMAKE_CXX_FLAGS_RELEASE "-O3")
 set(CMAKE_BUILD_TYPE release)
 
@@ -120,23 +134,25 @@ target_link_libraries(executorExampleAdvanced nvinfer_plugin_tensorrt_llm
                       cxxopts::cxxopts)
 
 # MultiInstance
+if(ENABLE_MULTI_DEVICE)
+  find_package(MPI REQUIRED)
+  message(STATUS "Using MPI_C_INCLUDE_DIRS: ${MPI_C_INCLUDE_DIRS}")
+  message(STATUS "Using MPI_C_LIBRARIES: ${MPI_C_LIBRARIES}")
+  include_directories(${MPI_C_INCLUDE_DIRS})
 
-find_package(MPI REQUIRED)
-message(STATUS "Using MPI_C_INCLUDE_DIRS: ${MPI_C_INCLUDE_DIRS}")
-message(STATUS "Using MPI_C_LIBRARIES: ${MPI_C_LIBRARIES}")
-include_directories(${MPI_C_INCLUDE_DIRS})
+  add_executable(executorExampleAdvancedMultiInstances
+                 executorExampleAdvancedMultiInstances.cpp)
+  target_link_libraries(
+    executorExampleAdvancedMultiInstances nvinfer_plugin_tensorrt_llm
+    cxxopts::cxxopts ${MPI_C_LIBRARIES})
 
-add_executable(executorExampleAdvancedMultiInstances
-               executorExampleAdvancedMultiInstances.cpp)
-target_link_libraries(
-  executorExampleAdvancedMultiInstances nvinfer_plugin_tensorrt_llm
-  cxxopts::cxxopts ${MPI_C_LIBRARIES})
+  # FastLogits
+  add_executable(executorExampleFastLogits executorExampleFastLogits.cpp)
+  target_link_libraries(executorExampleFastLogits nvinfer_plugin_tensorrt_llm
+                        cxxopts::cxxopts ${MPI_C_LIBRARIES})
 
-# FastLogits
-add_executable(executorExampleFastLogits executorExampleFastLogits.cpp)
-target_link_libraries(executorExampleFastLogits nvinfer_plugin_tensorrt_llm
-                      cxxopts::cxxopts ${MPI_C_LIBRARIES})
-
-add_executable(executorExampleDisaggregated executorExampleDisaggregated.cpp)
-target_link_libraries(executorExampleDisaggregated nvinfer_plugin_tensorrt_llm
-                      cxxopts::cxxopts ${MPI_C_LIBRARIES})
+  add_executable(executorExampleDisaggregated executorExampleDisaggregated.cpp)
+  target_link_libraries(
+    executorExampleDisaggregated nvinfer_plugin_tensorrt_llm cxxopts::cxxopts
+    ${MPI_C_LIBRARIES})
+endif()

--- a/scripts/build_cpp_examples.py
+++ b/scripts/build_cpp_examples.py
@@ -21,7 +21,7 @@ def working_directory(path: PathLike):
 
 
 def build_cpp_examples(build_dir: PathLike, trt_dir: PathLike,
-                       loglevel: int) -> None:
+                       enable_multi_device: str, loglevel: int) -> None:
     logging.basicConfig(level=loglevel,
                         format='%(asctime)s - %(levelname)s - %(message)s')
     # Convert input paths to pathlib.Path objects
@@ -52,6 +52,7 @@ def build_cpp_examples(build_dir: PathLike, trt_dir: PathLike,
             '-B',
             '.',
             f'-DTensorRT_ROOT={cmake_parse(trt_dir)}',
+            f'-DENABLE_MULTI_DEVICE={enable_multi_device}',
         ] + generator
         logging.info(f"Executing {generate_command}")
         subprocess.run(generate_command, check=True)
@@ -70,6 +71,9 @@ if __name__ == '__main__':
     parser.add_argument('--trt-dir',
                         default='/usr/local/tensorrt',
                         help='TensorRT directory path')
+    parser.add_argument('--enable-multi-device',
+                        default='ON',
+                        help='Enable multi device support (requires MPI)')
     parser.add_argument('-v',
                         '--verbose',
                         help="verbose",

